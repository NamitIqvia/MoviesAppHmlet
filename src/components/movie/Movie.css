
body{
    background-color: #7758e8;
}

.app-heading{
    color:white;
}

.search-box{
    width:350px;
    height:36px;
    border-radius:10px;
    border:2px solid #441cd6;
}

.search-box:focus{
    outline:none;
}

.movieDisplay{
    display:flex;
    flex-wrap:wrap;
    justify-content: center;
}
