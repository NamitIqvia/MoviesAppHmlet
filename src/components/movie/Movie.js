import React, { useEffect, useState } from 'react';
import './Movie.css';
import Moviegrid from './Moviegrid';


const searchApi = "https://api.themoviedb.org/3/search/movie?&api_key=e2dc6097770ffd325b2d2eceac46f1ce&query=";

const Movie = () => {
   
    const savedMovies = JSON.parse(localStorage.getItem('moviesdb'));
    const [currentMovie, setCurrentMovie] = useState("");
    const [searchMovie, setSearchMovie] = useState("")
    const [movies, setMovies] = useState(savedMovies || []);
    const [showLoading, setShowLoading] = useState(false);
    const [noMovieStatus, setNoMovieStatus] = useState(false);

     const getMovies = async () => {
        if(searchMovie){
        const response = await fetch(searchApi + String(searchMovie))
        let moviesList = await response.json();
        console.log(moviesList.results)
        setMovies(moviesList.results);
        localStorage.setItem('moviesdb', JSON.stringify(moviesList.results))
        if(moviesList.results.length === 0){
            setNoMovieStatus(true)
        }
        else{
            setNoMovieStatus(false)
        }
        setShowLoading(false)
     }
    }

     const addMovies = (e) => {
        setCurrentMovie(e.target.value)
    }

     const preventDefaultSubmit = (e) => {
        e.preventDefault();
        setShowLoading(true)
        setSearchMovie(currentMovie)
        setCurrentMovie("")
    }

    useEffect(() => {
        getMovies(searchMovie)
    }, [searchMovie])

    return (
        <>
         <h1 className="app-heading">Find your favourite movies below!!!</h1>
         <form onSubmit={(e) => {preventDefaultSubmit(e)}}>
           <input type="text" data-testid="search bar" value={currentMovie} onChange={(e) => {addMovies(e)}} className="search-box" placeholder="Enter the movie name here....."  />
         </form>

         <div>
             {showLoading && <div><h2>Loading...</h2></div>}
         </div>

         <div>
             {noMovieStatus && <div><h2>No movies found...</h2></div>}
         </div>

         <div className="movieDisplay">
             {
                 ((movies !== undefined) &&(movies.length > 0))  ? movies.map((movie) => 
                 <Moviegrid data={movie} key={movie.id}/>) : null
             }
         </div>

       </>
    )
}

export default Movie;
