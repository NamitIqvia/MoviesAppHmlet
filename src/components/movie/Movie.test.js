import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import  Movie  from './Movie';

/*it("search bar renders correctly", () => {
    const {queryByTestId} = render(<Movie />)
    expect(queryByTestId("search bar")).toBeTruthy()
})*/

afterEach(cleanup)

it('should take a snapshot', () => {
    const { asFragment } = render(<Movie />)
    expect(asFragment(<Movie />)).toMatchSnapshot()
});