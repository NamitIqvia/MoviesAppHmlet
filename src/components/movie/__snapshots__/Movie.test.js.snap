// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <h1
    class="app-heading"
  >
    Find your favourite movies below!!!
  </h1>
  <form>
    <input
      class="search-box"
      data-testid="search bar"
      placeholder="Enter the movie name here....."
      type="text"
      value=""
    />
  </form>
  <div />
  <div />
  <div
    class="movieDisplay"
  />
</DocumentFragment>
`;
